### DISCUSSION TITLE
file_browser插件在win11 24h2上后台是会报错的。win10 家庭版、专业版上没得问题

### DISCUSSION THREAD

**Discussion by yanghaoynu** [dyphire/mpv-config#143] [General]
file_browser插件在win11 24h2上后台是会报错的。win10 家庭版、专业版上没得问题

**Comment by dyphire** [comment]
错误日志呢？现在流行赛博诊断吗

**Reply by yanghaoynu** [reply]
谢谢大佬，已经解决了。问题是win11 24h2默认删除了系统组件wmic，https://win11.ithome.com/archiver/781/543.htm。而file_broser组件是使用这个win命令去获取磁盘分区信息的。我看了插件源码，在win11 24h2上建议用户按照下面的方法改进。
<img width="905" alt="Snipaste_2024-12-13_02-16-38" src="https://github.com/user-attachments/assets/3ec17953-917f-4ce2-8423-b3e6613d2167" />
1、设置--可选功能-中安装wmic
2、win11默认修改了powershell路径，深度用户一样会报错，因为没有pwsh命令，按照下图修改源码
<img width="1097" alt="Snipaste_2024-12-13_02-17-59" src="https://github.com/user-attachments/assets/654fd861-3bf9-41f6-b676-898a73e42faf" />
最后，也希望作者能在更新说明中做一个备注说明。

**Reply by dyphire** [reply]
> 谢谢大佬，已经解决了。问题是win11 24h2默认删除了系统组件wmic，[https://win11.ithome.com/archiver/781/543.htm。而file_broser组件是使用这个win命令去获取磁盘分区信息的

这个还真不知道，尝试把对应的代码改为`args = {'fsutil', 'fsinfo', 'drives'}`看下是否正常

> win11默认修改了powershell路径，深度用户一样会报错，因为没有pwsh命令，按照下图修改源码

open_dialog.lua 脚本中默认做了 powershell 的回退，按理来说不会出错，把报错内容发一下看看

**Reply by yanghaoynu** [reply]
1、果然还得是你，第一个问题按照你的参数，完美
2、ctrl+o键可以打开，但后台报错，且默认打开的是system32目录，可否调整？
\
<img width="1074" alt="Snipaste_2024-12-14_17-16-18" src="https://github.com/user-attachments/assets/1e20af6e-e47e-498d-95c7-f502f30084f5" />
<img width="953" alt="Snipaste_2024-12-14_17-15-55" src="https://github.com/user-attachments/assets/33ef36b3-3ad3-4e43-bdd5-376792c73523" />

**Reply by dyphire** [reply]
> 2、ctrl+o 键可以打开，但后台报错，且默认打开的是 system32 目录，可否调整？

控制台的 warn 信息是预期内的，对话框可以正常打开就表明 fallback 功能有正常生效。至于显示信息乱码是 mpv 的限制无法处理
对话框默认打开的目录行为是由 Windows 系统控制的，脚本无法调整。但打开过一次文件后就会记忆打开目录了

**Reply by yanghaoynu** [reply]
好的，谢谢大佬。

**Comment by dyphire** [comment]
在提交 dc30f83 中修复 wmic 相关问题

**Reply by yanghaoynu** [reply]
作者您好：我今天尝试了下把memus.lua的参数改成了args = {'fsutil', 'fsinfo', 'drives'}，后台不报错，但是驱动器没法正常显示。还是的用wmic。
<img width="1252" alt="Snipaste_2024-12-16_15-15-21" src="https://github.com/user-attachments/assets/e2bc21c6-2e0a-4169-9fa1-8931499299cc" />
<img width="953" alt="Snipaste_2024-12-16_15-15-40" src="https://github.com/user-attachments/assets/014f7de1-55f9-4a6f-b10d-4179dfd6a1a4" />

**Reply by dyphire** [reply]
因为 uosc 中的相关代码实现不太一样，需要照着我之前的提交 d2e5572 来改
